lib/tasks/lumen.rake

```diff
@@ -781,3 +738,90 @@ where works.id in (
     puts "#{date_time_task.call} Finishing the task"
   end
 end
+
+class CourtOrderReporter
+  def report
+    setup_directories
+    initialize_info_file
+    fetch_files
+    write_files
+    make_archive
+    email_user
+  end
+
+  def setup_directories
+    Rails.logger.info '[rake] Generating court order attachments report'
+
+    @magic_dir = ENV['USER_CRON_MAGIC_DIR'] || 'usercron'
+    @base_dir = Rails.root.join('public', @magic_dir)
+    Dir.mkdir(@base_dir) unless Dir.exist?(@base_dir)
+
+    @working_dir = Rails.root.join('public', @magic_dir, 'working')
+    Dir.mkdir(@working_dir) unless Dir.exist?(@working_dir)
+  end
+
+  def initialize_info_file
+    @info_filepath = File.join(@working_dir, 'info.txt')
+    info_header = <<~HEREDOC
+      The following notices (if any) have attached files which must be reviewed
+      by Lumen staff before they can be released:
+    HEREDOC
+
+    File.write(@info_filepath, info_header)
+  end
+
+  def fetch_files
+    @files = FileUpload.where(
+      notice: CourtOrder.where('created_at > ?', 1.week.ago),
+    )
+  end
+
+  def write_files
+    @files.each do |f|
+      if f.kind == 'supporting'
+        copy_file_to_archive(f)
+      else
+        notify_about_unredacted_files(f)
+      end
+    end
+  end
+
+  def copy_file_to_archive(f)
+    # The first two params ensure the filename is useful; the third ensures
+    # it is unique.
+    name = "#{f.notice_id}_#{f.id}_#{f.file_file_name}".gsub(/\s+/, "")
+    system("cp #{Shellwords.escape(f.file.path)} '#{File.join(@working_dir, name)}'")
+  end
+
+  def notify_about_unredacted_files(f)
+    return if f.notice.file_uploads.where(kind: 'supporting').present?
+    File.write(@info_filepath, app.notice_url(f.notice), mode: 'a')
+  end
+
+  def make_archive
+    Rails.logger.info '[rake] Making court order reports archive'
+    @archive_filename = Date.today.iso8601
+    system("tar -czvf #{File.join(@base_dir, @archive_filename)}.tar.gz -C #{@working_dir} .")
+    system("rm -r #{@working_dir}")
+  end
+
+  def email_user
+    email = ENV['USER_CRON_EMAIL']
+    unless (email && defined? SMTP_SETTINGS)
+      Rails.logger.warn '[rake] Missing email or SMTP_SETTINGS; not emailing court order report'
+      exit
+    end
+
+    Rails.logger.info '[rake] Sending court order report email'
+    mailtext = <<~HEREDOC
+      Subject: Latest email archive from Lumen
+
+      The latest archive of Lumen court order files can be found at
+      #{ Chill::Application.config.site_host}/#{@magic_dir}/#{@archive_filename}.tar.gz.
+    HEREDOC
+
+    Net::SMTP.start(SMTP_SETTINGS[:address]) do |smtp|
+      smtp.send_message mailtext, 'no-reply@lumendatabase.org', email
```

Comment:
Is there a way to verify this address? Gmail cannot verify this address and thus it is not able to distinguish if lumen actually sent this mail or a spammer

Reply:
If we start hearing significant user complaints about emails not reaching them, we'll open a ticket for antispam provisions. We know the relevant user has been successfully receiving these emails.